internal/util/git/git_test.go

```diff
@@ -370,3 +374,119 @@ func TestHTTPAuthMethodFile(t *testing.T) {
 	assert.Equal(t, "someuser", basicAuth.Username)
 	assert.Equal(t, "my-password", basicAuth.Password)
 }
+
+// NOTE: FetchOptions and PullOptions would be tested similarly except that the URL field would be RemoteURL instead.
```

Comment:
I guess we can run the same test cases with different options. but it is not necessary I assume.